src/site/views/ticketpurchase/tmpl/summary.php

```diff
@@ -201,52 +237,172 @@ class="swa-option-selector"
-	            e.preventDefault();
-
-                // Close Checkout on page navigation:
-                window.addEventListener('popstate', function () {
-                    handler.close();
-                });
-            } else {
-                document.getElementById('stripe-button').addEventListener('click', function (e) {
-                    jQuery('#stripe-form').submit();
-                })
-            }
-        });
 
+<!-- handle payment -->
+<script type="text/javascript">
+	// A reference to Stripe.js initialized with your real test publishable API key.
+	var stripe = Stripe("<?php echo $jConfig->get('stripe_publishable_key'); ?>");
+
+	// Disable the button until we have Stripe set up on the page
+	document.querySelector("#stripe-button").disabled = true;
+
+	var elements = stripe.elements();
+	var style = {
+		base: {
+			color: "#32325d",
+			fontFamily: 'Arial, sans-serif',
+			fontSmoothing: "antialiased",
+			fontSize: "16px",
+			"::placeholder": {
+				color: "#32325d"
+			}
+		},
+		invalid: {
+			fontFamily: 'Arial, sans-serif',
+			color: "#fa755a",
+			iconColor: "#fa755a"
+		}
+	};
+	var card = elements.create("card", {
+		style: style
+	});
+	// Stripe injects an iframe into the DOM
+	card.mount("#card-element");
+	card.on("change", function(event) {
+		// Disable the Pay button if there are no card details in the Element or add on not valid
+		$stripeCardErrorMsg = event.error ? event.error.message : "";
+		$cardDetailsComplete = event.complete;
+		setStripeButtonStatus();
+	});
+	var form = document.getElementById("payment-form");
+
+	var payWithCard = function(stripe, card, clientSecret) {
+		// console.log(stripe, card, clientSecret)
+		loading(true);
+		stripe
+			.confirmCardPayment(clientSecret, {
+				payment_method: {
+					card: card
+				}
+			})
+			.then(function(result) {
+				console.log(result)
+				if (result.error) {
+					// Show error to your customer
+					showError(result.error.message);
+				} else {
+					// The payment succeeded!
+					processOrder(result.paymentIntent.id);
+				}
+			})
+			.catch(function(error) {
+				// Handle the error
+				console.error("Error here");
+			});
+	};
+	/* ------- UI helpers ------- */
+	// Shows a success message when the payment is complete
+	var processOrder = function(paymentIntentId) {
+		loading(true);
+		document.querySelector(".result-message").classList.remove("hidden");
+		document.querySelector("#stripe-button").disabled = true;
+		fetch("<?php echo JRoute::_('index.php??option=com_swa&task=ticketpurchase.addTicketToDb') ?>", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json"
+				},
+				body: JSON.stringify({
+					paymentIntentId: paymentIntentId,
+				})
+			})
+			.then(function(result) {
+				if (result.ok) {
+					window.location.href = "<?php echo JRoute::_('index.php?option=com_swa&view=membertickets') ?>";
+				} else {
+					return result.text().then(text => {
+						throw new Error(text)
```

Comment:
I need a way to raise this to the user clearly and so they do not attempt to try again. The ShowError function diplays a message in the stripe UI element and lets the user try again. However an error is thrown here, the ticket may have been paid for and not added to the DB, therefore i need a more serious error to show, maybe on a new page

Reply:
@addshore @oshotton

Reply:
A quick a dirty solution could be to remove/hide the Stripe div from the page and just show the error message so the user can't try again?